packages/compile-solidity/index.js

```diff
@@ -74,6 +46,10 @@ const Compile = {
 
   // this takes an array of paths and options
   async sourcesWithDependencies({ paths, options }) {
+    if (options.compilers.solc.version === "analyzePragmas") {
```

Comment:
It will, but that should never be the case unless it is explicitly set that way. The `Config` object has defaults for that value [here](https://github.com/trufflesuite/truffle/blob/develop/packages/config/src/configDefaults.ts#L46-L57).